app/src/main/java/com/keylesspalace/tusky/util/ThemeUtils.java

```diff
@@ -104,29 +99,30 @@ public static void setDrawableTint(Context context, Drawable drawable, @AttrRes
     }
 
     public static void setAppNightMode(String flavor, Context context) {
-        int mode;
         switch (flavor) {
             default:
             case THEME_NIGHT:
-                mode = UiModeManager.MODE_NIGHT_YES;
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
                 break;
             case THEME_DAY:
-                mode = UiModeManager.MODE_NIGHT_NO;
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
                 break;
             case THEME_BLACK:
-                mode = UiModeManager.MODE_NIGHT_YES;
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
                 break;
             case THEME_AUTO:
-                mode = UiModeManager.MODE_NIGHT_AUTO;
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_AUTO);
                 break;
-        }
+            case THEME_SYSTEM:
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            UiModeManager uiModeManager = (UiModeManager)context.getApplicationContext().getSystemService(Context.UI_MODE_SERVICE);
-            uiModeManager.setNightMode(mode);
-        } else {
-            AppCompatDelegate.setDefaultNightMode(mode);
+                //stupid workaround to make MODE_NIGHT_FOLLOW_SYSTEM work :(
+                if((Settings.System.getInt(context.getContentResolver(), "display_night_theme", 0) == 1)) {
+                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES);
+                } else if ((Settings.System.getInt(context.getContentResolver(), "display_night_theme", 0) == 0)) {
+                    AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO);
```

Comment:
Why does this work? If you switch the system theme, Tusky won't follow if you set anything other than `MODE_NIGHT_FOLLOW_SYSTEM` or what am I missing?

Reply:
It then works because of workaround part 2 from there: https://github.com/tuskyapp/Tusky/pull/1069/files#r260898481

E.g. when switching from Tusky in Light theme (System is in Dark Theme) to Tusky in Auto theme it would stay in light until I switch System Theme to light and back to dark or kill the Tusky App and start it again, not sure if only Bug on Samsung S9 or whatever... That's why I check for system settings, set correct theme, apply it, then go back to auto theme, this way it works for me as expected. :/

Do you have a better suggestion there?